### ISSUE TITLE
Allow communication through proxy servers

### ISSUE THREAD

**Issue by boothj5** [profanity-im/profanity#365]
A few request have been made to allow profanity to proxy through HTTP etc for use in the workplace.

**Comment by temerick** [comment]
I would also be interested in this feature

**Comment by xeverse** [comment]
wtf no proxy facility what-so-ever?

**Comment by jubalh** [comment]
Yeah I also hate it when people offer their free time to create free software and then don't implement *everything* and not *perfectly* and not *immediately*..

**Comment by xeverse** [comment]
Yet they supposedly do 
it because they can; the path
coz i hope they are passionate 
about this & that & warm 
fuzzy feeling of course..

it's just proxy is 
such basic functionality 
so sastounding not to see it..
it's been almost 5y -- this thread.

> when people offer their free as in freedom time to create FOSS and 
> then don't implement _everything_ and not _perfectly_ and not _immediately_..

**Comment by jubalh** [comment]
> it's just proxy is
such basic functionality

Please create a pull request then.

**Comment by mdosch** [comment]
If you want proxy for using TOR:

```
cat /usr/local/bin/chat
#! /bin/bash
torify /usr/local/bin/profanity
```
Works well enough for me.

**Comment by qorg11** [comment]
According to Tor Project [torify should not be used](https://gitlab.torproject.org/legacy/trac/-/wikis/doc/torsocks). So it would be better to do something like this
```sh
alias profanity="torsocks -q profanity"
```

**Comment by jubalh** [comment]
@sjaeckel this depends on libstrophe, right?

**Comment by GasparVardanyan** [comment]
> I think maybe socks should be included alrso, right now I am using proxychains for this and it works great.
> 
> The thing is in pidgin for example libpurple supports the proxy protocol, not the client itself.

@brzezinki please, can you share your proxychains config?

**Comment by TURING-V2** [comment]
have similar problem more of a curl problem, my https upload server is on tor , and using torsocks and proxychains gives cant resolve hostname,  
So using HTTPS_PROXY=socks5h://localhost:9050  profanity , gets `failed: SSL peer certificate or SSH remote key was not OK` cause its self signed I need to pass accept cert or insecure (-k) from outside profanity .
Other way is to just use http.
